File: to-do app/to-do-go/src/controller/task_controller.go
package controller

import (
	"encoding/json"
	"errors"
	"io/ioutil"
	"net/http"
	"strconv"

	"github.com/gorilla/mux"
	"renan.com/todo/src/business"
	"renan.com/todo/src/controller/form"
	"renan.com/todo/src/controller/response"
	"renan.com/todo/src/security"
)

// GetTasksAndLists implements the controller to get tasks and lists
func GetTasksAndLists(w http.ResponseWriter, r *http.Request) {
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	tasksListsDTO, bErr := business.GetTasksAndLists(userID)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, tasksListsDTO, r)
}

// GetTasksByList implements the controller to get tasks by a selected list
func GetTasksByList(w http.ResponseWriter, r *http.Request) {
	param := mux.Vars(r)
	listID, err := strconv.ParseUint(param["id"], 10, 64)
	if err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	taskDTOs, bErr := business.GetTasksByList(userID, listID)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, taskDTOs, r)
}

// GetTasksOnDefaultList implements the controller to get tasks not related to a list
func GetTasksOnDefaultList(w http.ResponseWriter, r *http.Request) {
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	taskDTOs, bErr := business.GetTasksByList(userID, 0)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, taskDTOs, r)
}

// CreateTaskList implements the controller to create a new lists
func CreateTaskList(w http.ResponseWriter, r *http.Request) {
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	reqBody, err := ioutil.ReadAll(r.Body)
	if err != nil {
		response.Err(w, http.StatusUnprocessableEntity, err, r)
		return
	}
	var form form.TaskListForm
	if err = json.Unmarshal(reqBody, &form); err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	taskListDTO, bErr := business.CreateTaskList(form.ListName, form.ListDescription, userID)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusCreated, taskListDTO, r)
}

// UpdateTaskList implements the controller to update a list data
func UpdateTaskList(w http.ResponseWriter, r *http.Request) {
	param := mux.Vars(r)
	listID, err := strconv.ParseUint(param["id"], 10, 64)
	if err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	reqBody, err := ioutil.ReadAll(r.Body)
	if err != nil {
		response.Err(w, http.StatusUnprocessableEntity, err, r)
		return
	}
	var form form.TaskListForm
	if err = json.Unmarshal(reqBody, &form); err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	bErr := business.UpdateTaskList(listID, userID, form.ListName, form.ListDescription)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, nil, r)
}

// DeleteTaskList implements the controller to delete a list and it's tasks
func DeleteTaskList(w http.ResponseWriter, r *http.Request) {
	param := mux.Vars(r)
	listID, err := strconv.ParseUint(param["id"], 10, 64)
	if err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	bErr := business.DeleteTaskList(listID, userID)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, nil, r)
}

// MoveTasksFromList implements the controller to change tasks from one list to another
func MoveTasksFromList(w http.ResponseWriter, r *http.Request) {
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	reqBody, err := ioutil.ReadAll(r.Body)
	if err != nil {
		response.Err(w, http.StatusUnprocessableEntity, err, r)
		return
	}
	var form form.MoveTasksFromListForm
	if err = json.Unmarshal(reqBody, &form); err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	bErr := business.MoveTasksFromList(userID, form.ListIDOrigin, form.ListIDDestiny)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, nil, r)
}

// CompleteTasksFromList implements the controller to change an array of tasks status to complete
func CompleteTasksFromList(w http.ResponseWriter, r *http.Request) {
	param := mux.Vars(r)
	listID, err := strconv.ParseUint(param["id"], 10, 64)
	if err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	bErr := business.CompleteTasksFromList(userID, listID)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, nil, r)
}

// UncompleteTasksFromList implements the controller to change an array of tasks status to incomplete
func UncompleteTasksFromList(w http.ResponseWriter, r *http.Request) {
	param := mux.Vars(r)
	listID, err := strconv.ParseUint(param["id"], 10, 64)
	if err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	bErr := business.UncompleteTasksFromList(userID, listID)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, nil, r)
}

// CreateTask implements the controller to create a new task
func CreateTask(w http.ResponseWriter, r *http.Request) {
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	reqBody, err := ioutil.ReadAll(r.Body)
	if err != nil {
		response.Err(w, http.StatusUnprocessableEntity, err, r)
		return
	}
	var form form.TaskForm
	if err = json.Unmarshal(reqBody, &form); err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	taskDTO, bErr := business.CreateTask(form.TaskName, form.TaskDescription, form.Deadline, userID, form.ListID)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusCreated, taskDTO, r)
}

// UpdateTask implements the controller to update a task
func UpdateTask(w http.ResponseWriter, r *http.Request) {
	param := mux.Vars(r)
	id, err := strconv.ParseUint(param["id"], 10, 64)
	if err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	reqBody, err := ioutil.ReadAll(r.Body)
	if err != nil {
		response.Err(w, http.StatusUnprocessableEntity, err, r)
		return
	}
	var form form.TaskForm
	if err = json.Unmarshal(reqBody, &form); err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	bErr := business.UpdateTask(form.TaskName, form.TaskDescription, form.Deadline, userID, id)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, nil, r)
}

// MoveTaskToList implements the controller to update a task list (move one task to other list)
func MoveTaskToList(w http.ResponseWriter, r *http.Request) {
	param := mux.Vars(r)
	id, err := strconv.ParseUint(param["id"], 10, 64)
	if err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	reqBody, err := ioutil.ReadAll(r.Body)
	if err != nil {
		response.Err(w, http.StatusUnprocessableEntity, err, r)
		return
	}
	var form form.MoveTaskToListForm
	if err = json.Unmarshal(reqBody, &form); err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	bErr := business.MoveTaskToList(userID, form.ListID, id)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, nil, r)
}

// DeleteTask implements the controller to delete a task
func DeleteTask(w http.ResponseWriter, r *http.Request) {
	param := mux.Vars(r)
	id, err := strconv.ParseUint(param["id"], 10, 64)
	if err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	bErr := business.DeleteTask(userID, id)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, nil, r)
}

// CompleteTask implements the controller to update a task status to complete
func CompleteTask(w http.ResponseWriter, r *http.Request) {
	param := mux.Vars(r)
	id, err := strconv.ParseUint(param["id"], 10, 64)
	if err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	bErr := business.CompleteTask(userID, id)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, nil, r)
}

// UncompleteTask implements the controller to update a task status to incomplete
func UncompleteTask(w http.ResponseWriter, r *http.Request) {
	param := mux.Vars(r)
	id, err := strconv.ParseUint(param["id"], 10, 64)
	if err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusUnauthorized, err, r)
		return
	}
	bErr := business.UncompleteTask(userID, id)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, nil, r)
}

File: to-do app/to-do-java/src/main/java/com/renan/todo/controller/TaskController.java
package com.renan.todo.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RestController;

import com.renan.todo.business.BusinessException;
import com.renan.todo.business.TaskService;
import com.renan.todo.controller.filter.AuthFilter;
import com.renan.todo.controller.form.MoveTaskToListForm;
import com.renan.todo.controller.form.MoveTasksFromListToListForm;
import com.renan.todo.controller.form.TaskForm;
import com.renan.todo.controller.form.TaskListForm;
import com.renan.todo.dto.ErrorDTO;
import com.renan.todo.dto.TaskDTO;
import com.renan.todo.dto.TaskListDTO;
import com.renan.todo.util.JwtUtil;
import com.renan.todo.util.MessageUtil;

/**
 * The controller for tasks and lists operations
 */
@RestController()
@SuppressWarnings("rawtypes")
public class TaskController {

	@Autowired
	private TaskService taskService;

	/**
	 * Fetch the lists and tasks for the user
	 * 
	 * @param jwt - the auth token from the user
	 * @return - the ResponseEntity
	 */
	@GetMapping(value = "/api/taskList", produces = MediaType.APPLICATION_JSON_VALUE)
	public ResponseEntity getTasksAndLists(@RequestHeader(AuthFilter.AUTH) String jwt) {
		try {
			Integer           userID = JwtUtil.getIdUser(jwt);
			List<TaskListDTO> lists  = taskService.getTasksAndLists(userID);
			return ResponseEntity.ok(lists);
		} catch (BusinessException e) {
			return e.getResponseEntity();
		}
	}

	/**
	 * Create a new List for the user
	 * 
	 * @param jwt  - the auth token from the user
	 * @param form - form data for new task list
	 * @return - the ResponseEntity
	 */
	@PostMapping(value = "/api/taskList", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
	public ResponseEntity createTaskList(@RequestHeader(AuthFilter.AUTH) String jwt, @RequestBody TaskListForm form) {
		try {
			if (form.getListName() == null) {
				return ResponseEntity.status(HttpStatus.BAD_REQUEST)
				        .body(new ErrorDTO(MessageUtil.getErrorMessageInputValues()));
			}
			Integer     userID   = JwtUtil.getIdUser(jwt);
			TaskListDTO taskList = taskService.createTaskList(form.getListName().trim(),
			        form.getListDescription() != null ? form.getListDescription().trim() : null, userID);
			return ResponseEntity.ok(taskList);
		} catch (BusinessException e) {
			return e.getResponseEntity();
		}
	}

	/**
	 * Update a TaskList
	 * 
	 * @param jwt    - the auth token of the user, to get user data
	 * @param form   - the input from the user
	 * @param idList - list identifier
	 * @return - Response Entity
	 */
	@PutMapping(value = "/api/taskList/{id}", consumes = MediaType.APPLICATION_JSON_VALUE)
	public ResponseEntity updateTaskList(@RequestHeader(AuthFilter.AUTH) String jwt, @RequestBody TaskListForm form,
	        @PathVariable("id") Integer idList) {
		try {
			if (form.getListName() == null || idList == null || idList <= 0) {
				return ResponseEntity.status(HttpStatus.BAD_REQUEST)
				        .body(new ErrorDTO(MessageUtil.getErrorMessageInputValues()));
			}
			Integer userID = JwtUtil.getIdUser(jwt);
			taskService.updateTaskList(form.getListName().trim(),
			        form.getListDescription() != null ? form.getListDescription().trim() : null, idList, userID);
			return ResponseEntity.ok().build();
		} catch (BusinessException e) {
			return e.getResponseEntity();
		}
	}

	/**
	 * Delete a TaskList and the tasks related
	 * 
	 * @param jwt    - the auth token of the user, to get user data
	 * @param idList - list identifier
	 * @return - Response Entity
	 */
	@DeleteMapping(value = "/api/taskList/{id}")
	public ResponseEntity deleteTaskList(@RequestHeader(AuthFilter.AUTH) String jwt,
	        @PathVariable("id") Integer idList) {
		try {
			if (idList == null || idList <= 0) {
				return ResponseEntity.status(HttpStatus.BAD_REQUEST)
				        .body(new ErrorDTO(MessageUtil.getErrorMessageInputValues()));
			}
			Integer userID = JwtUtil.getIdUser(jwt);
			taskService.deleteTaskList(idList, userID);
			return ResponseEntity.ok().build();
		} catch (BusinessException e) {
			return e.getResponseEntity();
		}
	}

	/**
	 * Complete all tasks from the list passed as parameter
	 * 
	 * @param jwt    - the auth token of the user, to get user data
	 * @param idList - list identifier
	 * @return - Response Entity
	 */
	@PutMapping(value = "/api/taskList/tasks/complete/{id}")
	public ResponseEntity completeTaskList(@RequestHeader(AuthFilter.AUTH) String jwt,
	        @PathVariable("id") Integer idList) {
		try {
			if (idList == null || idList <= 0) {
				return ResponseEntity.status(HttpStatus.BAD_REQUEST)
				        .body(new ErrorDTO(MessageUtil.getErrorMessageInputValues()));
			}
			Integer userID = JwtUtil.getIdUser(jwt);
			taskService.completeTaskList(idList, userID);
			return ResponseEntity.ok().build();
		} catch (BusinessException e) {
			return e.getResponseEntity();
		}
	}

	/**
	 * Complete all tasks from the list passed as parameter
	 * 
	 * @param jwt    - the auth token of the user, to get user data
	 * @param idList - list identifier
	 * @return - Response Entity
	 */
	@PutMapping(value = "/api/taskList/tasks/uncomplete/{id}")
	public ResponseEntity uncompleteTaskList(@RequestHeader(AuthFilter.AUTH) String jwt,
	        @PathVariable("id") Integer idList) {
		try {
			if (idList == null || idList <= 0) {
				return ResponseEntity.status(HttpStatus.BAD_REQUEST)
				        .body(new ErrorDTO(MessageUtil.getErrorMessageInputValues()));
			}
			Integer userID = JwtUtil.getIdUser(jwt);
			taskService.uncompleteTaskList(idList, userID);
			return ResponseEntity.ok().build();
		} catch (BusinessException e) {
			return e.getResponseEntity();
		}
	}

	/**
	 * Create a new task for the user
	 * 
	 * @param jwt  - the auth token from the user
	 * @param form - form data for new task list
	 * @return - the ResponseEntity
	 */
	@PostMapping(value = "/api/task", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
	public ResponseEntity createTask(@RequestHeader(AuthFilter.AUTH) String jwt, @RequestBody TaskForm form) {
		try {
			if (form.getTaskName() == null) {
				return ResponseEntity.status(HttpStatus.BAD_REQUEST)
				        .body(new ErrorDTO(MessageUtil.getErrorMessageInputValues()));
			}
			Integer userID = JwtUtil.getIdUser(jwt);
			TaskDTO task   = taskService.createTask(form.getTaskName().trim(),
			        form.getTaskDescription() != null ? form.getTaskDescription().trim() : null,
			        form.getDeadline() != null ? form.getDeadline().trim() : null,
			        form.getIdList() != null && form.getIdList() > 0 ? form.getIdList() : null, userID);
			return ResponseEntity.ok(task);
		} catch (BusinessException e) {
			return e.getResponseEntity();
		}
	}

	/**
	 * Update task for the user
	 * 
	 * @param jwt    - the auth token from the user
	 * @param form   - form data for new task list
	 * @param taskID - task id
	 * @return - the ResponseEntity
	 */
	@PutMapping(value = "/api/task/{id}", consumes = MediaType.APPLICATION_JSON_VALUE)
	public ResponseEntity updateTask(@RequestHeader(AuthFilter.AUTH) String jwt, @RequestBody TaskForm form,
	        @PathVariable("id") Integer taskID) {
		try {
			if (form.getTaskName() == null || taskID == null || taskID <= 0) {
				return ResponseEntity.status(HttpStatus.BAD_REQUEST)
				        .body(new ErrorDTO(MessageUtil.getErrorMessageInputValues()));
			}
			Integer userID = JwtUtil.getIdUser(jwt);
			taskService.updateTask(taskID, form.getTaskName().trim(),
			        form.getTaskDescription() != null ? form.getTaskDescription().trim() : null,
			        form.getDeadline() != null ? form.getDeadline().trim() : null, userID);
			return ResponseEntity.ok().build();
		} catch (BusinessException e) {
			return e.getResponseEntity();
		}
	}

	/**
	 * Delete task for the user
	 * 
	 * @param jwt    - the auth token from the user
	 * @param taskID - task id
	 * @return - the ResponseEntity
	 */
	@DeleteMapping(value = "/api/task/{id}")
	public ResponseEntity deleteTask(@RequestHeader(AuthFilter.AUTH) String jwt, @PathVariable("id") Integer taskID) {
		try {
			if (taskID == null || taskID <= 0) {
				return ResponseEntity.status(HttpStatus.BAD_REQUEST)
				        .body(new ErrorDTO(MessageUtil.getErrorMessageInputValues()));
			}
			Integer userID = JwtUtil.getIdUser(jwt);
			taskService.deleteTask(taskID, userID);
			return ResponseEntity.ok().build();
		} catch (BusinessException e) {
			return e.getResponseEntity();
		}
	}

	/**
	 * Mark task as completed for the user
	 * 
	 * @param jwt    - the auth token from the user
	 * @param taskID - task id
	 * @return - the ResponseEntity
	 */
	@PutMapping(value = "/api/task/complete/{id}")
	public ResponseEntity completeTask(@RequestHeader(AuthFilter.AUTH) String jwt, @PathVariable("id") Integer taskID) {
		try {
			if (taskID == null || taskID <= 0) {
				return ResponseEntity.status(HttpStatus.BAD_REQUEST)
				        .body(new ErrorDTO(MessageUtil.getErrorMessageInputValues()));
			}
			Integer userID = JwtUtil.getIdUser(jwt);
			taskService.completeTask(taskID, userID);
			return ResponseEntity.ok().build();
		} catch (BusinessException e) {
			return e.getResponseEntity();
		}
	}

	/**
	 * Mark task as incompleted for the user
	 * 
	 * @param jwt    - the auth token from the user
	 * @param taskID - task id
	 * @return - the ResponseEntity
	 */
	@PutMapping(value = "/api/task/uncomplete/{id}")
	public ResponseEntity unCompleteTask(@RequestHeader(AuthFilter.AUTH) String jwt,
	        @PathVariable("id") Integer taskID) {
		try {
			if (taskID == null || taskID <= 0) {
				return ResponseEntity.status(HttpStatus.BAD_REQUEST)
				        .body(new ErrorDTO(MessageUtil.getErrorMessageInputValues()));
			}
			Integer userID = JwtUtil.getIdUser(jwt);
			taskService.unCompleteTask(taskID, userID);
			return ResponseEntity.ok().build();
		} catch (BusinessException e) {
			return e.getResponseEntity();
		}
	}

	/**
	 * Set the task to another list
	 * 
	 * @param jwt    - the auth token from the user
	 * @param form   - form data for the operation
	 * @param taskID - task id
	 * @return - the ResponseEntity
	 */
	@PutMapping(value = "/api/task/list/{id}", consumes = MediaType.APPLICATION_JSON_VALUE)
	public ResponseEntity moveTaskForList(@RequestHeader(AuthFilter.AUTH) String jwt,
	        @RequestBody MoveTaskToListForm form, @PathVariable("id") Integer taskID) {
		try {
			if (taskID == null || taskID <= 0 || form.getListId() == null || form.getListId() <= 0) {
				return ResponseEntity.status(HttpStatus.BAD_REQUEST)
				        .body(new ErrorDTO(MessageUtil.getErrorMessageInputValues()));
			}
			Integer userID = JwtUtil.getIdUser(jwt);
			taskService.moveTaskForList(taskID, form.getListId(), userID);
			return ResponseEntity.ok().build();
		} catch (BusinessException e) {
			return e.getResponseEntity();
		}
	}

	/**
	 * Move tasks from a List to another
	 * 
	 * @param jwt  - the auth token from the user
	 * @param form - form data for the operation
	 * @return - the ResponseEntity
	 */
	@PutMapping(value = "/api/taskList/tasks/moveFromList", consumes = MediaType.APPLICATION_JSON_VALUE)
	public ResponseEntity moveTaskFromAListToAnother(@RequestHeader(AuthFilter.AUTH) String jwt,
	        @RequestBody MoveTasksFromListToListForm form) {
		try {
			Integer userID = JwtUtil.getIdUser(jwt);
			taskService.moveTaskFromAListToAnother(form.getListIdOrigin(), form.getListIdDestiny(), userID);
			return ResponseEntity.ok().build();
		} catch (BusinessException e) {
			return e.getResponseEntity();
		}
	}

}
